configure.ac

```diff
@@ -7,14 +7,52 @@ AC_CONFIG_HEADERS([config.h])
 AC_PROG_CXX
 AM_PROG_AR
 
+AC_CANONICAL_HOST
+
+build_linux=no
+build_windows=no
+build_mac=no
+
+# Detect the target system
+case "${host_os}" in
+    linux*)
+        build_linux=yes
+        ;;
+    cygwin*|mingw*)
+        build_windows=yes
+        ;;
+    darwin*)
+        AC_MSG_ERROR(["OS $host_os is not supported"])
+        ;;
+    *)
+        AC_MSG_ERROR(["OS $host_os is not supported"])
+        ;;
+esac
+
+# Pass the conditionals to automake
+AM_CONDITIONAL([LINUX], [test "$build_linux" = "yes"])
+AM_CONDITIONAL([WINDOWS], [test "$build_windows" = "yes"])
+AM_CONDITIONAL([OSX], [test "$build_mac" = "yes"])
+
 AC_LANG([C++])
 LT_INIT
 
+# Windows mu have no undefined symbol in dll, we need -no-undefined 
+# flag to work, so lets trigger clean dll build
+
+AM_COND_IF( [WINDOWS], [
+LT_INIT([win32-dll])
+])
+
+
+
```

Comment:
Please don't add these blank lines as well as more below.  Ideally, just have a single blank line between different blocks of statements.

Reply:
Ok!

Reply:
Ok!